toucan_connectors/snowflake_common.py

```diff
@@ -178,6 +177,9 @@ def fetch_data(
         limit: Optional[int] = None,
         get_row_count: bool = False,
     ) -> pd.DataFrame:
+        self._execute_query(connection, f'USE DATABASE {data_source.database}')
+        self._execute_query(connection, f'USE WAREHOUSE {data_source.warehouse}')
```

Comment:
en supposant que _execute_query fait un call http, tu pourrais pas faire un seul appel en faisant : 

```suggestion
        self._execute_query(connection, f'USE DATABASE {data_source.database}; 'USE WAREHOUSE {data_source.warehouse}')
```

Reply:
Vu avec @piotch  pour intégrer le sujet dans une carte supplémentaire sur de l'optimisation